downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,599 @@
+<note>
+    <to>Tove</to>
+    <from>Jani</from>
+    <heading>Reminder</heading>
+    <body>Don't forget me this weekend!</body>
+</note>
+```
+##### Email Module Improvements
+
+- Make the `body` field of the `email:Message` record optional. This enables sending an email with only the `htmlBody` field set without a text-typed `body` field.
+
+- Add `mime:Entity` type to the union type of the `attachments` field in the `email:Message` record. That enables attaching a single MIME Entity as an attachment. This `email:Message` record change would appear as follows.
+
+```ballerina
+public type Message record {|
+    // ... other fields
+    mime:Entity|Attachment|(mime:Entity|Attachment)[] attachments?;
+|};
+```
+
+- The above-mentioned changes related to the `body` and `attachments` fields are updated in the `email:Options` record as given below in order to facilitate the same in the `sendEmail` method in the `email:SmtpClient`.
+
+```ballerina
+public type Options record {|
+    // ... other fields
+    string body?;
+    mime:Entity|Attachment|(mime:Entity|Attachment)[] attachments?;
+|};
+```
+
+- Remove the `properties` field representing custom properties from all Email module related configuration records: `email:SmtpConfig`, `email:PopConfig`, `email:ImapConfig`, `email:PopListenerConfig`, and `email:ImapListenerConfig`.
+
+- The `mail.smtp.ssl.checkserveridentity` custom property was passed as an entry in the `properties` to enable/disable the server certificate’s hostname verification. As the `properties` field is removed from the API, from this release onwards, the `verifyHostName` boolean field is introduced to the `secureSocket` record for all the configurations related to the Email module. 
+
+This `email:SecureSocket` record change would appear as follows.
+
+```ballerina
+public type SecureSocket record {|
+    // ... other fields
+    boolean verifyHostname = true;
+|};
+```
+
+##### TCP Module Improvements
+
+Introduced SSL/TLS-based communication to the TCP module using the `secureSocket` record configuration in both the client and the listener.
+
+###### TCP Client with `secureSocket`
+
+Sample code is as follows.
+
+```ballerina
+import ballerina/io;
+import ballerina/tcp;
+
+configurable string certPath = ?;
+
+public function main() returns error? {
+    tcp:Client socketClient = check new ("localhost", 9002, secureSocket = {
+        certificate: {path: certPath},
+        protocol: {
+            name: "TLS",
+            versions: ["TLSv1.2", "TLSv1.1"]
+        },
+        ciphers: ["TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"]
+    });
+    string msg = "Hello Ballerina Echo from secure client";
+    byte[] msgByteArray = msg.toBytes();
+    check socketClient->writeBytes(msgByteArray);
+    readonly & byte[] receivedData = check socketClient->readBytes();
+    io:println(string:fromBytes(receivedData));
+    check socketClient->close();
+}
+```
+
+###### TCP Listener with `secureSocket`
```

Comment:
Same comment as above regarding the use of `secureSocket` in the heading

Reply:
@chamil321

Reply:
@ldclakmal

Reply:
I think it is decided to remove this section. @MohamedSabthar

Reply:
Need to remove the `tcp module improvement` section since it only contains the secure socket implementation

Reply:
Removed.